### PR TITLE
Initial e2e tests setup for https://github.com/kodadot/nft-gallery/issues/1503

### PR DESCRIPTION
**Pull request by rchovatiya88**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Do a quick check before the merge.

### PR type

- [ ] Bugfix
- [x ] Feature
- [ ] Refactoring

### What's new?

- [ ] PR closes #<issue_number>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases

### Had issue bounty label ?

- [ ] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=<My_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address>)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix 
Untitled_ Jan 28, 2022 10_05 PM (1)
has changed **something** on UI, a screenshot is best to understand changes for others.

### CODE REVIEW

File: cypress/integration/e2e/search.spec.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable indent */
+
+/* eslint-disable no-undef */
+describe('Navigate to Gallery', () => {
+  it('Navigate to the Gallery page', () => {
+    cy.visit('http://localhost:9090/rmrk/gallery')
+    cy.wait(2000)
```
Comment (kkukelka) [inline cypress/integration/e2e/search.spec.js:7]: please don't use wait and instead try to watch for the response of initial graphql call via [cy.intercept](https://docs.cypress.io/api/commands/intercept)

File: cypress/integration/e2e/smokeTests.spec.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable indent */
+
+/* eslint-disable no-undef */
+const baseUrl = 'http://localhost:9090'
+describe('Navigate to Homepage ', () => {
+  it('Navigate to the Home page', () => {
+    cy.visit(baseUrl)
+    cy.wait(2000)
```
Comment (kkukelka) [inline cypress/integration/e2e/smokeTests.spec.js:8]: here as well

File: cypress/integration/e2e/search.spec.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  })
+
+  it('Search for term', () => {
+    cy.get('.input').type('generative', { force: true })
```
Comment (kkukelka) [inline cypress/integration/e2e/search.spec.js:11]: `.input` is a very generice selector. please query the element via [data-cy](https://docs.cypress.io/guides/references/best-practices#How-It-Works)

File: cypress/integration/e2e/search.spec.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  })
+
+  it('View Search Results', () => {
+    cy.get('.input').type('{enter}')
```
Comment (kkukelka) [inline cypress/integration/e2e/search.spec.js:15]: `.input` is a very generice selector. please query the element via [data-cy](https://docs.cypress.io/guides/references/best-practices#How-It-Works)

File: cypress/integration/e2e/search.spec.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    cy.get('.section').click({ force: true })
```
Comment (kkukelka) [inline cypress/integration/e2e/search.spec.js:16]: `.section` is a very generice selector. please query the element via [data-cy](https://docs.cypress.io/guides/references/best-practices#How-It-Works)

File: cypress/integration/e2e/smokeTests.spec.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  })
+
+  it('Search for term', () => {
+    cy.get('.input').type('generative', { force: true })
```
Comment (kkukelka) [inline cypress/integration/e2e/smokeTests.spec.js:12]: same as above

File: cypress/integration/e2e/smokeTests.spec.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  })
+
+  it('View Search Results', () => {
+    cy.get('.input').type('{enter}')
```
Comment (kkukelka) [inline cypress/integration/e2e/smokeTests.spec.js:16]: same as above

File: cypress/integration/e2e/smokeTests.spec.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    cy.get('.section').click({ force: true })
```
Comment (kkukelka) [inline cypress/integration/e2e/smokeTests.spec.js:17]: same as above

**Review by kkukelka — CHANGES_REQUESTED** [review]
lots of things to improve 🌮

**Comment by rchovatiya88** [comment]
Thanks!

This was first commit to make sure cypress is part of repo. So was jus example test 
 
You are correct and understood will def use data-cy or add selectors going forward 

Wait usage I argue to a point. 
I will need ur help in setting cy.intercept for graphql-- still a noob 

For now adding it and will create more tests accordingly 🥪

**Comment by kkukelka** [comment]
> Thanks!
> 
> This was first commit to make sure cypress is part of repo. So was jus example test
> 
> You are correct and understood will def use data-cy or add selectors going forward
> 
> Wait usage I argue to a point. I will need ur help in setting cy.intercept for graphql-- still a noob
> 
> For now adding it and will create more tests accordingly 🥪

I mean, if everyone is ok with it @yangwao @roiLeo we can merge it like that as a base setup and I will do a more precise rework in a separate PR.

**Comment by yangwao** [comment]
> I mean, if everyone is ok with it @yangwao @roiLeo we can merge it like that as a base setup and I will do a more precise rework in a separate PR.

Yup, let's move it to `/tests/cypress/` as I guess we will have `/tests/` directory already present :)

Let's merge it this stage then, no need to wait for me on this

**Review by kkukelka — APPROVED** [review]

**Comment by rchovatiya88** [comment]
@kkukelka if you would like to merge it, i'll create new branch with fixes n better tests :)

**Comment by kkukelka** [comment]
> @kkukelka if you would like to merge it, i'll create new branch with fixes n better tests :)

I'm not allowed to merge sadly 😛 and this branch still has merge conflicts in `yarn.lock`

**Comment by yangwao** [comment]
> @kkukelka if you would like to merge it, i'll create new branch with fixes n better tests :)

Let's merge this and you can make PR on top of this later :)

Like need to move `/cypress` directory under `/tests` and we are good to go :)

**Pull request merged**